### ISSUE TITLE
[Feat]: introduce getContextsWithDetails endpoint

### ISSUE THREAD

**Issue by ilendemli** [appium/appium#20036]
**Is your feature request related to a problem? Please describe.**

In a hybrid app when getting the available context handles you see available context from the remote debugger in the logs with bunch of informations:

```
[RemoteDebugger]     Application: "PID:4323"
[RemoteDebugger]         id: "PID:4323"
[RemoteDebugger]         isProxy: false
[RemoteDebugger]         name: "APP"
[RemoteDebugger]         bundleId: "some.app.id"
[RemoteDebugger]         hostId: undefined
[RemoteDebugger]         isActive: true
[RemoteDebugger]         isAutomationEnabled: "Unknown"
[RemoteDebugger]         pageArray:
[RemoteDebugger]           - id: 1
[RemoteDebugger]             title: "Some title"
[RemoteDebugger]             url: "https://example.com"
[RemoteDebugger]             isKey: true
[RemoteDebugger]     Application: "PID:632"
[RemoteDebugger]         id: "PID:632"
[RemoteDebugger]         isProxy: true
[RemoteDebugger]         name: "Safari"
[RemoteDebugger]         bundleId: "com.apple.SafariViewService"
[RemoteDebugger]         hostId: undefined
[RemoteDebugger]         isActive: true
[RemoteDebugger]         isAutomationEnabled: "Unknown"
[RemoteDebugger]         pageArray:
[RemoteDebugger]           - id: 22
[RemoteDebugger]             title: "PayPal Checkout"
[RemoteDebugger]             url: "https://www.sandbox.paypal.com/webapps/"
[RemoteDebugger]             isKey: false
[RemoteDebugger]           - id: 43
[RemoteDebugger]             title: ""
[RemoteDebugger]             url: "about:blank"
[RemoteDebugger]             isKey: false
```

Currently one has to switch context, get maybe the page source, can get the current URL if one exists, and is only able to access limited informations about the contexts.

**Describe the solution you'd like**

I would like appium to introduce a new endpoint to get the available contexts with details in one go, without switching the context first. Would also be great to pass on some parameter to filter the results for example if the WebView is Active, or if the page is not blank etc.

**Describe alternatives you've considered**

Current we are switching contexts to get the url and to check for it.

**Additional context**

**Comment by mykola-mokhnach** [comment]
Have you tried https://github.com/appium/appium-xcuitest-driver/blob/master/docs/reference/execute-methods.md#mobile-getcontexts ?

**Comment by ilendemli** [comment]
@mykola-mokhnach oh I did not see that, I have looked into `SupportsContextSwitching` and did not see the method there so I assumed there was no such feature. I will check this out.

**Comment by amedvedjev** [comment]
@ilendemli there is also capability
```
// https://appiumpro.com/editions/61-how-to-accurately-select-webviews-using-the-fullcontextlist-capability
.setFullContextList(true) 
```

**Comment by ilendemli** [comment]
Sorry for going silent, I am sick and not able to try this right now. But I tried `mobile: getContexts` before I went to sick-leave and this worked as expected. The field names differ between iOS and Android though.

@amedvedjev Thank you for the link. I remember `getContextHandles()` inside SupportsContextSwitching returning a list of strings. How does that work out here? I will try it as soon as I can.
